File: week1/cash.c
#include <stdio.h>

int main(void) 
{
    int cents;

  
    do 
    {
        printf("Change owed: ");
        scanf("%d", &cents);
    } 
    while (cents < 0);

    int coins = 0;

   
    while (cents >= 25) 
    {
        cents = cents - 25;
        coins++;
    }

  
    while (cents >= 10) 
    {
        cents = cents - 10;
        coins++;
    }

   
    while (cents >= 5) 
    {
        cents = cents - 5;
        coins++;
    }

    
    while (cents >= 1) 
    {
        cents = cents - 1;
        coins++;
    }

   
    printf("%d\n", coins);

    return 0;
}

File: week6/cash.py
from cs50 import get_float

def main():
    
    while True:
        dollars = get_float("Change owed: ")
        if dollars > 0:
            break
            
    cents = int(dollars * 100 + 0.5)
    
    coins = 0
    
    
    coins += cents // 25
    cents %= 25
    
    coins += cents // 10
    cents %= 10
    
    coins += cents // 5
    cents %= 5
    
    coins += cents // 1
    cents %= 1
    
    print(coins)

if __name__ == "__main__":
    main()